app/code/Magento/Deploy/Package/PackagePool.php

```diff
@@ -41,22 +46,46 @@ class PackagePool
      */
     private $collected = false;
 
+    /**
+     * @var StoreView|null
+     */
+    private $storeView;
+
+    /**
+     * @var UserCollection|null
+     */
+    private $userCollection;
+
+    /**
+     * @var Locale|null
+     */
+    private $locale;
+
     /**
      * PackagePool constructor
      *
      * @param Collector $collector
      * @param ListInterface $themeCollection
      * @param PackageFactory $packageFactory
+     * @param StoreView|null $storeView
+     * @param UserCollection|null $userCollection
+     * @param Locale|null $locale
      */
     public function __construct(
         Collector $collector,
         ListInterface $themeCollection,
-        PackageFactory $packageFactory
+        PackageFactory $packageFactory,
+        ?StoreView $storeView = null,
+        ?UserCollection $userCollection = null,
+        ?Locale $locale = null
     ) {
         $this->collector = $collector;
         $themeCollection->clear()->resetConstraints();
         $this->themes = $themeCollection->getItems();
         $this->packageFactory = $packageFactory;
+        $this->storeView = $storeView ?: ObjectManager::getInstance()->get(StoreView::class);
+        $this->userCollection = $userCollection ?: ObjectManager::getInstance()->get(UserCollection::class);
```

Comment:
collection should not be used in constructor as this object will be shared with other places and state will be kept that will lead to unexpected issues. 
Instead we need to use collection factory that will be automatically generated for the collection

Reply:
The userCollection is now a factory